src/primitives/sphere.jl

```diff
@@ -37,6 +37,30 @@ end
 
 Sphere(p1::Tuple, p2::Tuple, p3::Tuple) = Sphere(Point(p1), Point(p2), Point(p3))
 
+"""
+    Sphere(p1, p2, p3, p4)
+
+A 3D sphere passing through points `p1`, `p2`, `p3` and `p4`.
+"""
+function Sphere(p1::Point{3}, p2::Point{3}, p3::Point{3}, p4::Point{3})
+  v1 = p1 - p4
+  v2 = p2 - p4
+  v3 = p3 - p4
+  d = det(hcat(v1, v2, v3))
```

Comment:
This line represents the volume of a geometry? Can you write it in terms of volume so that more code gets reused? Also, probably a good idea to rename the result to vol

Reply:
Ah no you're right. This determinant is `1/6` of the volume.

Reply:
https://mathinsight.org/relationship_determinants_area_volume

Reply:
Yes, you are basically checking if the four points lie on the same plane. This is equivalent to checking if the tetrahedron has non-zero volume. So the entire code can be reduced to something like:

```julia
if volume(Tetrahedron(p1, p2, p3, p4)) > 0
  # do something
else
  # throw error
end
```